specs/deneb/beacon-chain.md

```diff
@@ -232,6 +239,22 @@ def kzg_commitment_to_versioned_hash(kzg_commitment: KZGCommitment) -> Versioned
     return VERSIONED_HASH_VERSION_KZG + hash(kzg_commitment)[1:]
 ```
 
+#### `get_max_blobs_per_block`
+
+```python
+def get_max_blobs_per_block(epoch: Epoch) -> uint64:
+    """
+    Return the maximum number of blobs that can be included in a block for a given epoch.
+    """
+    assert len(BLOB_SCHEDULE) > 0
+    for entry in sorted(BLOB_SCHEDULE, key=lambda e: e["EPOCH"], reverse=True):
+        if epoch >= entry["EPOCH"]:
+            return entry["MAX_BLOBS_PER_BLOCK"]
+
+    # Only for testing. Should never reach this in the real world.
+    return min(entry["MAX_BLOBS_PER_BLOCK"] for entry in BLOB_SCHEDULE)
```

Comment:
wouldn't it be better to use max blobs of the smallest epoch instead? Since `BLOB_SCHEDULE` can have entries that decrease max blobs it might be unexpected if such a entry is picked for unscheduled epochs

Reply:
Eh it shouldn't really matter. This line will only be executed in tests.

Reply:
was mostly thinking about kurtosis testing, if you override the blob schedule it's possible to reach this line as well, but might not be worth to change this now

Reply:
The real issue with this line is that under the minimal preset, it will return 6 for all forks (including Electra and Fulu). But there's really not a good solution to this other than revamping the testing infrastructure so that minimal uses realistic fork epoch values.

Reply:
I think the fork epochs in minimal doesn't have to be the same as mainnet, but they should be different numbers not all the same. Why are all the same?

It worries me that we add this because people will read this, in general, not as something for testing but prescription for their clients. The comment fixes this, but it would be neater to don't have things in the spec that the clients should implement other way.

Another separate thing:

What about adding to the blob schedule `(Epoch:0, MAX_BLOBS_PER_BLOCK: 0)` it is a more elegant way to represent that there are no blobs before.